misc/git_pick.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+Help="Given a Git rev-id and list of regular expressions $0 searches \
+branch starting from rev-id for commits whose 'git log' lines matches at \
+least one regular expression given. \
+Selected commit hashes are printed to stdout, other messages to stderr.\n\
+\n\
+Example:\n\n\
+git cherry-pick \$($0 branch_name \\\\\n\
+'feature regular expression' \\\\\n\
+'author name' \\\\\n\
+'commit message prefix' \\\\\n\
+2>>/dev/null)\n\
+"
+
+Usage="usage: $0 [-h] [--help] rev-id regex [regex [...]]"
+
+
```

Comment:
No args? This is caught by consequent logic. And this is an error to run the script without args (`exit 1`). Full help is not required without explicit request.

Reply:
The current message is not entirely clear. Without specifying an rev-id, an empty string is used, and the error text says that the log was not found, not that the rev-id was not specified.
> fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
no log for rev-id
usage: ./git_pick.sh [-h] [--help] rev-id string [string [...]]

For example, running a `qemu_device_creator.py` without arguments will produce:
> usage: qemu_device_creator.py [-h]
                              [--qemu-build /path/to/qemu/build/directory]
                              [--target-version <tree-ish>]
                              [--gen-header-tree header_tree.gv]
                              [--gen-chunk-graphs]
                              [--gen-intermediate-chunk-graphs]
                              [--gen-debug-comments] [--no-i3s]
                              [--no-instruction-tree-optimizations]
                              script
qemu_device_creator.py: error: the following arguments are required: script

It might be possible to explicitly check for the presence of a rev-id:
```
if [ 0 == $# ]
then
	echo "no rev-id" 1>&2
	echo "$Usage" 1>&2
	exit 1
fi

LogRevId=$1
shift
```